conformance/tests/mesh-consumer-route.yaml

```diff
@@ -0,0 +1,17 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: mesh-echo-add-header
+  namespace: gateway-conformance-mesh-consumer
+spec:
+  parentRefs:
+  - kind: Service
+    name: echo-v1
+    namespace: gateway-conformance-mesh
+  rules:
+  - filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: X-Header-Set
+          value: set
```

Comment:
This relies on https://gateway-api.sigs.k8s.io/geps/gep-1426/?h=xro#omitting-backendrefs which is only SHOULD. Can we put the `backendRef` explicitly?